src/python/Utils/Throttled.py

```diff
@@ -16,41 +36,131 @@
 
 class _ThrottleCounter(object):
     """
-    _ThrottleCounter class define throttle parameter and
+    _ThrottleCounter class defines throttle parameter and
     enter/exit methods to work with `with` context.
     """
 
     def __init__(self, throttle, user, debug=False):
         self.throttle = throttle
         self.user = user
-        self.debug = debug
+        if debug:
+            self.throttle.logger.setLevel(logging.DEBUG)
 
     def __enter__(self):
         "Define enter method for `with` context"
         ctr = self.throttle._incUser(self.user)
-        if self.debug:
-            msg = "Entering throttled function with counter %d for user %s" \
-                    % (ctr, self.user)
-            self.throttle.logger.debug(msg)
+	msg = "Entering throttled function with counter %d for user %s" \
+		% (ctr, self.user)
+	self.throttle.logger.debug(msg)
         if ctr >= self.throttle.getLimit():
             self.throttle._decUser(self.user)
```

Comment:
I still think we should not decrement the counter here. It's already getting decremented when it leaves the context manager. 
E.g.: limit is set to 3 and user put a 4th request in the system, it enters the context manager and gets incremented to 4, then it finds it has exceeded the limit, decrements back to 3, leave the context manager and decrements it back to 2.

Another thing I just noticed, if limit is 3, then we should allow a 3rd call, this line doesn't:
`if ctr >= self.throttle.getLimit():`
it should be just greater than, not greater or equal than.

Reply:
fixed